homeassistant/components/eheimdigital/fan.py

```
@@ -0,0 +1,135 @@
+
+    coordinator = entry.runtime_data
+
+    def async_setup_device_entities(
+        device_address: str | dict[str, EheimDigitalDevice],
+    ) -> None:
+        """Set up the climate entities for one or multiple devices."""
+        entities: list[EheimDigitalClassicVarioPump] = []
+        if isinstance(device_address, str):
+            device_address = {device_address: coordinator.hub.devices[device_address]}
+        for device in device_address.values():
+            if isinstance(device, EheimDigitalClassicVario):
+                entities.append(EheimDigitalClassicVarioPump(coordinator, device))
+                coordinator.known_devices.add(device.mac_address)
+
+        async_add_entities(entities)
+
+    coordinator.add_platform_callback(async_setup_device_entities)
+
+    async_setup_device_entities(coordinator.hub.devices)
+
+
+class EheimDigitalClassicVarioPump(
+    EheimDigitalEntity[EheimDigitalClassicVario], FanEntity
+):
+    """Represent an EHEIM Digital filter pump."""
+
+    _attr_supported_features = (
+        FanEntityFeature.PRESET_MODE
+        | FanEntityFeature.SET_SPEED
+        | FanEntityFeature.TURN_OFF
+        | FanEntityFeature.TURN_ON
+    )
+    _attr_preset_modes = [FILTER_BIO_MODE, FILTER_PULSE_MODE]
+    _attr_preset_mode: str | None = None
+    _attr_translation_key = "filter"
+    _attr_name = None
+
+    def __init__(
+        self,
+        coordinator: EheimDigitalUpdateCoordinator,
+        device: EheimDigitalClassicVario,
+    ) -> None:
+        """Initialize an EHEIM Digital classicVARIO light entity."""
+        super().__init__(coordinator, device)
+        self._attr_unique_id = self._device_address
+        self._async_update_attrs()
+
+    @override
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set the preset mode."""
+        try:
+            if preset_mode in FILTER_PRESET_TO_FILTER_MODE:
```

We should raise a `ServiceValidationError` when they provide an unsupported/invalid `preset_mode`.

---

This is already done inside the `fan` component, so it is not needed in the single integrations.